samples/snippets/src/main/java/pubsublite/ProducerExample.java

```diff
@@ -55,17 +57,22 @@ public static void producerExample(
     ProducerSettings producerSettings =
         ProducerSettings.newBuilder().setTopicPath(topicPath).build();
 
-    for (long i = 0L; i < 10L; i++) {
-      String key = "demo";
-      try (Producer<byte[], byte[]> producer = producerSettings.instantiate()) {
-        Future<RecordMetadata> sent =
+    List<Future<RecordMetadata>> futures = new ArrayList<>();
+    try (Producer<byte[], byte[]> producer = producerSettings.instantiate()) {
+      for (long i = 0L; i < 10L; i++) {
+        String key = "demo";
+        Future<RecordMetadata> future =
             producer.send(
                 new ProducerRecord(
                     topicPath.toString(), key.getBytes(), ("message-" + i).getBytes()));
-        RecordMetadata meta = sent.get();
-        System.out.println(meta.offset());
+        futures.add(future);
       }
     }
+
+    for (Future<RecordMetadata> future : futures) {
```

Comment:
This needs to be inside the try block: You terminated the stream associated with the publisher after you left the try block.

Reply:
I moved it inside the try block. Odd that the tests still passed when I ran them from my machine.